code/app/src/androidTest/java/com/kernelcrew/moodapp/HomeFeedNavigationTest.java

```diff
@@ -7,48 +7,47 @@
 
 import android.os.SystemClock;
 
+import androidx.test.espresso.Espresso;
 import androidx.test.espresso.action.ViewActions;
+import androidx.test.espresso.matcher.ViewMatchers;
 import androidx.test.ext.junit.rules.ActivityScenarioRule;
 import androidx.test.ext.junit.runners.AndroidJUnit4;
 
-import com.google.firebase.firestore.CollectionReference;
-import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
 import com.kernelcrew.moodapp.ui.MainActivity;
-import com.kernelcrew.moodapp.ui.Mood;
 
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
+import java.util.concurrent.ExecutionException;
+
 @RunWith(AndroidJUnit4.class)
 public class HomeFeedNavigationTest extends FirebaseEmulatorMixin {
+    // Changeable strings
+    private static final String TEST_EMAIL = "test@kernelcrew.com";
+    private static final String TEST_PASSWORD = "Password@1234";
+
     @Rule
     public ActivityScenarioRule<MainActivity> activityScenarioRule =
             new ActivityScenarioRule<>(MainActivity.class);
 
     @Before
-    public void seedDatabase() {
-        // Seed the Firestore "moods" collection with a sample document.
-        // This ensures that when HomeFeed loads, it has at least one mood document.
-        FirebaseFirestore db = FirebaseFirestore.getInstance();
-        CollectionReference moodsRef = db.collection("mood");
-
-        // Create a test mood using the correct constructor parameters:
-        // id, userName, moodText, timestamp.
-        Mood testMood = new Mood("testMoodId", "dummyUser", "Test mood", System.currentTimeMillis());
-        moodsRef.document("testMoodId").set(testMood);
-
-        // Wait for the data to be written
-        SystemClock.sleep(2000);
+    public void setUp() throws InterruptedException, ExecutionException {
+        createUser(TEST_EMAIL, TEST_PASSWORD);
+        SystemClock.sleep(1000);
```

Comment:
Instead of sleeping, you can call `Tasks.await` on the `createUserTask`. That should be much less flaky -- and, in this case, faster.